kong/pdk/tracing.lua

```diff
@@ -195,17 +195,13 @@ local function create_span(tracer, options)
     sampled = tracer and tracer.sampler(trace_id)
   end
 
-  if not sampled then
-    return noop_span
```

Comment:
returning noop meant that the next span being created would have no parent, so it would be created with a new trace-id and going through the sampler with a new trace-id means there is a chance the child could get sampled while the parent was not.

Reply:
I searched to find that `is_recording` decides whether to propagate the span (which is not very close to its name).
Is there a reason we need to have another field other than `should_sample`? Maybe we could merge them into 1 field.

Reply:
they are 2 slightly different things, is_recording is [part of the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isrecording) to indicate a span that is not accepting any data. You are right though that in this case it would be more appropriate to check for the value of `should_sample` to take propagation decisions.